apps/charterafrica/src/lib/tools/github.js

```diff
@@ -0,0 +1,77 @@
+import fetchJson, { FetchError } from "@/charterafrica/utils/fetchJson";
+
+export const GET_REPOSITORY = `query($repositoryOwner: String!, $repositoryName: String!) {
+  repository(owner: $repositoryOwner, name: $repositoryName) {
+    name
+    url
+    description
+    stargazers {
+      totalCount
+    }
+    watchers {
+      totalCount
+    }
+    forks {
+      totalCount
+    }
+    languages(first: 15) {
+      nodes {
+        name
+      }
+    }
+    defaultBranchRef {
+      target {
+        ... on Commit {
+          history(first: 1) {
+            edges {
+              node {
+                id
+                author {
+                  name
+                  email
+                }
+                committedDate
+                message
+              }
+            }
+          }
+        }
+      }
+    }
+    collaborators(first: 100) {
+      nodes {
+        login
+        name
+        bio
+        location
+        twitterUsername
+      }
+    }
+    owner {
+      ... on Organization {
+        login
+        name
+        description
+        location
+        twitterUsername
+      }
+    }
+  }
+}`;
+
+export const fetchRepository = async (variables) => {
+  const url = "https://api.github.com/graphql";
+  const headers = {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${process.env.GITHUB_TOKEN}`,
+  };
+  const data = {
+    variables,
+    query: GET_REPOSITORY,
+  };
+  const res = await fetchJson.post(url, { data, headers });
+  if (res?.data?.repository) {
+    return res.data.repository;
+  }
+  throw new FetchError("An Error occured", res.errors, 400);
```

Comment:
Why are we assuming it's a 400?

Reply:
Because Gitghub graphql errors' response status is 200. So this was default code since all other error codes have been handled by fetchJson

Reply:
But 400 is a user/client-side error. Can a user do anything about this?

Reply:
User may have entered a wrong github repository e.g `CodeForAfrica/non-existent-repository`. 
We may also default to `500`.

Reply:
I don't think users enter anything into the system... content team, yes. Is

1. Throwing error the best approach, or
2. Just logging it somewhere (Sentry?) so that we know there is any issue with a given row?

Reply:
At the moment, I am throwing all the errors wherever they occur.
All these errors are caught on controller level. Response object has a list of successful processes and failed processes.